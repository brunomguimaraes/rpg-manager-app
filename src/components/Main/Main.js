// @flow

import React from "react";
import { Switch, Redirect, Route } from "react-router-dom";
import Dashboard from "~/components/Dashboard";
import GameManager from "~/components/GameManager";
import styles from "./Main.module.scss";

export default function Main() {
  return (
    <div className={styles.main}>
      <Switch>
        <Route path="/dashboard" exact component={Dashboard} />
        <Route path="/game" exact component={GameManager} />
        <Route path="/game/:id" exact component={GameManager} />
        <Redirect from="/" to="/dashboard" />
      </Switch>
    </div>
  );
}
