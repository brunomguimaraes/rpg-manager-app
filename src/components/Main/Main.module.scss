@import "~styles/variables";

.main {
  height: 100%;
  display: flex;
  box-sizing: border-box;
  background: $light-primary;
}
