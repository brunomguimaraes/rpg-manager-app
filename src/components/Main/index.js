// @flow

export { default } from "./Main";
