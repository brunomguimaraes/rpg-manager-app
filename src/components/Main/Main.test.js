import React from "react";
import Main from "./Main";

test("Main", () => expect(shallow(<Main />)).toMatchSnapshot());
