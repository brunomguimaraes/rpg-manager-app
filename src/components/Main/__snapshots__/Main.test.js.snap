// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main 1`] = `
<div
  className="main"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/dashboard"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/game/:gameId?"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/game/:gameId/character/:characterId?"
    />
    <Redirect
      from="/"
      push={false}
      to="/dashboard"
    />
  </Switch>
</div>
`;
